Specifications/EventHorizon/Consumer/for_Subscriptions/when_subscribing/and_there_is_no_configuration_for_microservice.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using Machine.Specifications;
+
+namespace Dolittle.Runtime.EventHorizon.Consumer.for_Subscriptions.when_subscribing
+{
+    public class and_there_is_no_configuration_for_microservice : given.all_dependencies
+    {
+        static SubscriptionResponse result;
+        Because of = () => result = subscriptions.Subscribe(subscription_id with
+        {
+            ProducerMicroserviceId = Guid.Parse("565fafa6-d0ef-4662-96a3-b291abffb0fc")
+        }).GetAwaiter().GetResult();
```

Comment:
I don't think `with` is a C# keyword for mutating a record? Also this should be moved to the`Establish` block too, here's my example of what it could look like, GitHub comment boxes aren't the best for coding.

```suggestion
    public class and_there_is_no_configuration_for_microservice : given.all_dependencies
    {
        static SubscriptionResponse result;
        static SubscriptionId unknown_producer_subscription;
        
        Establish context = () => unknown_producer_subscription = new(
                    "aa6403de-6dee-4db8-80a0-366bb9532b3a",
                    // unknown producer microservice id
                    Guid.Parse("565fafa6-d0ef-4662-96a3-b291abffb0fc"),
                    "1da8200b-6da6-44fc-a442-29210fb5b934",
                    "51355173-6c09-4efa-bd4f-122c6cbc67fc",
                    Guid.Parse("cd2433d3-4aad-41ac-87bc-f37240736444"),
                    "064749fc-8d13-4b12-a3c4-584e34596b99");

        Because of = () => result = subscriptions.Subscribe(unknown_producer_subscription).GetAwaiter().GetResult();
```

Reply:
Nice catch, yes it should be in Establish. For the `with` comment, it does not mutate a record (they are immutable by default) it simply creates a copy of that record 'with' other specified values in specified fields